public/js/experiment.js

```diff
@@ -0,0 +1,101 @@
+"use strict";
+
+/* global ga */
+
+function selectURL(e, skipAnalyticsPing = false) {
+  const shareModalInput = document.getElementById("shareModalInput");
+
+  // If an a user-init'd focus, send analytics ping
+  if (!skipAnalyticsPing) {
+  sendShareModalPing(shareModalInput);
+  }
+
+  shareModalInput.select();
+  shareModalInput.setSelectionRange(0, 99999);
+}
+
+
+function copyURL(e) {
+  sendShareModalPing(e.target);
+  const shareModalInput = document.getElementById("shareModalInput");
+  // Remove select listener after this event is fired. (Dup engagement)
+  shareModalInput.removeEventListener("focus", selectURL);
+  selectURL(null, true);
+  document.execCommand("copy");
+}
+
+function keyPress(e) {
+  if(e.key === "Escape") {
+    closeShareModal();
+  document.removeEventListener("keydown", keyPress);
+  }
+}
+
+function closeShareModal() {
+  const shareModal = document.getElementById("shareModal");
+  shareModal.style.display = "none";
+}
+
+function sendShareModalPing(el) {
+  if (typeof(ga) !== "undefined") {
+    const eventCategory = "[v2] exp5-share-modal";
+    const eventAction = el.dataset.eventAction;
+    const eventLabel =  `Link ID: ${el.dataset.eventLabel}`;
+    const options = {};
+    return ga("send", "event", eventCategory, eventAction, eventLabel, options);
+  }
+}
+
+function initShareModal(target, breachText) {
+  const shareModal = document.getElementById("shareModal");
+  const shareModalInput = document.getElementById("shareModalInput");
+  const shareModalCopy = document.getElementById("shareModalCopy");
+  const btnCloseShareModal = document.getElementById("closeShareModal");
+  const shareTextBreach = document.getElementById("shareTextBreach");
+  const shareTextGeneral = document.getElementById("shareTextGeneral");
+
+  const shareModalLocation = target.dataset.eventLabel;
+
+  shareModalCopy.dataset.eventLabel = shareModalLocation;
+  shareModalInput.dataset.eventLabel = shareModalLocation;
+
+  // Set input to correct URL value;
+  shareModalInput.value = target.href;
+
+  // Make modal visible
+  shareModal.style.display = "block";
+
+  if (breachText) {
+  shareTextGeneral.classList.add("hidden");
+  shareTextBreach.classList.remove("hidden");
+  } else {
+  shareTextGeneral.classList.remove("hidden");
+  shareTextBreach.classList.add("hidden");
+  }
```

Comment:
nit: indenting

I should see if ESLint is catching these on CI.

Reply:
Doesn't look like ESLint is enforcing indentation. If I enable the rule w/ `"indent": ["warn", 2],`, we get the following output:

> "✖ 607 problems (0 errors, 607 warnings)"

(But I also disabled that `"no-warning-comments"` rule to suppress those 19 warnings so your mileage may vary.)

Reply:
RIP https://github.com/mozilla/blurts-server/pull/1700 💀

Reply:
Added `/*eslint indent: ["error", 2]*/` to the top of the file and ran my ESLinter (and then removed that line).